Source/PINCaching.h

```diff
@@ -0,0 +1,185 @@
+ and executes the passed block after the object has been stored, potentially in parallel with other blocks
+ on the <concurrentQueue>.
+ 
+ @param object An object to store in the cache.
+ @param key A key to associate with the object. This string will be copied.
+ @param cost An amount to add to the <memoryCache.totalCost>.
+ @param block A block to be executed concurrently after the object has been stored, or nil.
+ */
+- (void)setObjectAsync:(id)object forKey:(NSString *)key withCost:(NSUInteger)cost completion:(nullable PINCacheObjectBlock)block;
+
+/**
+ Removes the object for the specified key. This method returns immediately and executes the passed
+ block after the object has been removed, potentially in parallel with other blocks on the <concurrentQueue>.
+ 
+ @param key The key associated with the object to be removed.
+ @param block A block to be executed concurrently after the object has been removed, or nil.
+ */
+- (void)removeObjectForKeyAsync:(NSString *)key completion:(nullable PINCacheObjectBlock)block;
+
+/**
+ Removes all objects from the cache that have not been used since the specified date. This method returns immediately and
+ executes the passed block after the cache has been trimmed, potentially in parallel with other blocks on the <concurrentQueue>.
+ 
+ @param date Objects that haven't been accessed since this date are removed from the cache.
+ @param block A block to be executed concurrently after the cache has been trimmed, or nil.
+ */
+- (void)trimToDateAsync:(NSDate *)date completion:(nullable PINCacheBlock)block;
+
+/**
+ Removes all objects from the cache.This method returns immediately and executes the passed block after the
+ cache has been cleared, potentially in parallel with other blocks on the <concurrentQueue>.
+ 
+ @param block A block to be executed concurrently after the cache has been cleared, or nil.
+ */
+- (void)removeAllObjectsAsync:(nullable PINCacheBlock)block;
+
+
+#pragma mark - Synchronous Methods
+/// @name Synchronous Methods
+
+/**
+ This method determines whether an object is present for the given key in the cache.
+ 
+ @see containsObjectForKey:block:
+ @param key The key associated with the object.
+ @result YES if an object is present for the given key in the cache, otherwise NO.
+ */
+- (BOOL)containsObjectForKey:(NSString *)key;
+
+/**
+ Retrieves the object for the specified key. This method blocks the calling thread until the object is available.
+ Uses a lock to achieve synchronicity on the disk cache.
+ 
+ @see objectForKey:block:
+ @param key The key associated with the object.
+ @result The object for the specified key.
+ */
+- (nullable id)objectForKey:(NSString *)key;
+
+/**
+ Stores an object in the cache for the specified key. This method blocks the calling thread until the object has been set.
+ Uses a lock to achieve synchronicity on the disk cache.
+ 
+ @see setObject:forKey:block:
```

Comment:
Can you update these comments to point to the new async versions? There appear to be a few more in this file.